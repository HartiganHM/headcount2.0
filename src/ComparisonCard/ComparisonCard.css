@import url("https://fonts.googleapis.com/css?family=Open+Sans|Roboto+Slab:700");
.comparison-card {
  align-items: stretch;
  border: 1px solid #6D6E71;
  border-radius: 1rem;
  box-shadow: 0.5rem 0.5rem 0.2rem #6D6E71;
  box-sizing: border-box;
  cursor: default;
  display: flex;
  flex-direction: column;
  height: 18rem;
  overflow: hidden;
  justify-content: space-between;
  margin: 1rem;
  text-align: center;
  max-width: 12rem;
  min-width: 12rem; }
  .comparison-card .district-data, .comparison-card .comparison-data {
    box-sizing: border-box;
    height: 4rem;
    padding: 1rem; }
  .comparison-card .district-data {
    color: #FFF;
    background-color: #029fb4;
    font-size: 1rem; }
  .comparison-card .comparison-data {
    color: #00cc44;
    font-size: 0.8rem; }
