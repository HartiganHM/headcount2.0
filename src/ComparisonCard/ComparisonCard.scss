@import '../variables.scss';

.comparison-card {
    align-items: stretch;
    border: 1px solid gray;
    border-radius: 1rem;
    box-shadow: 0.5rem 0.5rem 0.2rem $main-color-dark;
    display: flex;
    flex-direction: column;
    height: 18rem;
    overflow: hidden;
    justify-content: space-between;
    margin: 1rem;
    text-align: center;
    max-width: 12rem;
    min-width: 12rem;

    .district-data, .comparison-data {
        box-sizing: border-box;
        height: 4rem;
        padding: 1rem;
    }

    .district-data {
        color: $white;
        background-color: $accent-color;
        font-size: 1rem;
    }

    .comparison-data {
        color: $high-score;
        font-size: 0.8rem;
    }
}