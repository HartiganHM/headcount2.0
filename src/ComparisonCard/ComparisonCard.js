import React from 'react';
import PropTypes from 'prop-types';
import './ComparisonCard.css';

const ComparisonCard = ({ comparedData }) => {
    const keys = Object.keys(comparedData);
    const data = keys.map( (district) => {
        return comparedData[district]
    })

    const district1 = keys.length ? `${keys[0]}:${data[0]}` : 'District 1';
    const district2 = keys.length ? `${keys[1]}:${data[1]}` : 'District 2';
    const comparison = keys.length ? `${keys[2]}:${data[2]}` : 'COMPARISON';

    return(
        <div className='comparison-card'>
            <span>{district1}</span>
            <span>{comparison}</span>
            <span>{district2}</span>
        </div>
    )
}

export default ComparisonCard