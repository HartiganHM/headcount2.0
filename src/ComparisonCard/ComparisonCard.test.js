import React from 'react';
import ComparisonCard from './ComparisonCard';
import { shallow } from 'enzyme';

describe('Comparison Card Tests', () => {
  let renderedComparisonCard;
  let mockData;

  beforeEach(() => {
    mockData = {};
    renderedComparisonCard = shallow(
      <ComparisonCard comparedData={mockData} />
    );
  });

  it('Comparison Card should exist', () => {
    expect(renderedComparisonCard).toBeDefined();
  });

  it('Comparison Card should match the snapshot', () => {
    expect(renderedComparisonCard).toMatchSnapshot();
  });

  it('Comparison Card should have default value if data does not exist', () => {
    expect(renderedComparisonCard.find('.comparison-data').text()).toEqual('-');
  });

  it('Comparison Card changes if data exists', () => {
    mockData = { d1: 0.1, d2: 0.3, compared: 0.2 };
    renderedComparisonCard = shallow(
      <ComparisonCard comparedData={mockData} />
    );
    expect(renderedComparisonCard.find('.comparison-data').text()).toEqual(
      'compared: 0.2'
    );
  });
});
