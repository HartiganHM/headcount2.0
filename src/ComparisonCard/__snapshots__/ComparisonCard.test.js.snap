exports[`Comparison Card Tests Comparison Card should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ComparisonCard
    comparedData={Object {}} />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <span
          className="district-data">
          Pick two districts
</span>,
        <span
          className="comparison-data">
          <i
                    className="icon-left-big" />
          -
          <i
                    className="icon-right-big" />
</span>,
        <span
          className="district-data">
          to compare statistics
</span>,
      ],
      "className": "comparison-card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Pick two districts",
          "className": "district-data",
        },
        "ref": null,
        "rendered": "Pick two districts",
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <i
              className="icon-left-big" />,
            "-",
            <i
              className="icon-right-big" />,
          ],
          "className": "comparison-data",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "icon-left-big",
            },
            "ref": null,
            "rendered": null,
            "type": "i",
          },
          "-",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "icon-right-big",
            },
            "ref": null,
            "rendered": null,
            "type": "i",
          },
        ],
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "to compare statistics",
          "className": "district-data",
        },
        "ref": null,
        "rendered": "to compare statistics",
        "type": "span",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span
            className="district-data">
            Pick two districts
</span>,
          <span
            className="comparison-data">
            <i
                        className="icon-left-big" />
            -
            <i
                        className="icon-right-big" />
</span>,
          <span
            className="district-data">
            to compare statistics
</span>,
        ],
        "className": "comparison-card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Pick two districts",
            "className": "district-data",
          },
          "ref": null,
          "rendered": "Pick two districts",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <i
                className="icon-left-big" />,
              "-",
              <i
                className="icon-right-big" />,
            ],
            "className": "comparison-data",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "icon-left-big",
              },
              "ref": null,
              "rendered": null,
              "type": "i",
            },
            "-",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "icon-right-big",
              },
              "ref": null,
              "rendered": null,
              "type": "i",
            },
          ],
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "to compare statistics",
            "className": "district-data",
          },
          "ref": null,
          "rendered": "to compare statistics",
          "type": "span",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
